viewing_party/party.py

```diff
@@ -1,23 +1,152 @@
 # ------------- WAVE 1 --------------------
 
 def create_movie(title, genre, rating):
-    pass
+    new_movie = {}
+    if not title or not genre or not rating:
+        return None
+    new_movie["title"] = title
+    new_movie["genre"] = genre
+    new_movie["rating"] = rating
+    return new_movie
+
+
+def add_to_watched(user_data, movie):
+    movies_watched_list = user_data["watched"]
+    movies_watched_list.append(movie)
+    return user_data
+
+
+def add_to_watchlist(user_data, movie):
+    movies_to_watch_list = user_data["watchlist"]
+    movies_to_watch_list.append(movie)
+    return user_data
+
+
+def watch_movie(user_data, title):
+    movies_to_watch_list = user_data["watchlist"]
+    movies_watched_list = user_data["watched"]
+
+    i = 0
+    for movie in movies_to_watch_list:
+        if movie["title"] == title:
+            movies_watched_list.append(movie)
+            del movies_to_watch_list[i]
+        i+=1

```

Comment:
Instead of using `i` as a counter we can simplify this by deleting the movie from the watchlist. Similarly to line 32, something like this:
```
def watch_movie(user_data, title):
    for movie in user_data["watchlist"]:
        if movie["title"] == title:
            user_data["watchlist"].remove(movie)
            user_data["watched"].append(movie)

    return user_data
```
Or this:
```
def watch_movie(user_data, title):
    for movie in movies_to_watch_list:
        if movie["title"] == title:
            user_data["watchlist"].remove(movie)
            movies_watched_list.append(movie)

    return user_data